[RDY] Doc cleanup (3) + terminal remnants

Pyrohh opened the pull request.

justinmk commented on runtime/doc/vim_diff.txt:
```diff
@@ -114,6 +114,10 @@ Other options:
+  'ttybuiltin'
+  'ttymouse'
+  'ttyscroll'
+  'weirdinvert'
```
should probably start sorting these alphanumerically

justinmk commented on runtime/doc/todo.txt:
```diff
@@ -546,11 +546,6 @@ is confusing.  Should say "the argument list is empty".
-URXVT:
-- will get stuck if byte sequence does not contain the expected semicolon.
-- Use urxvt mouse support also in xterm.  Explanations:
-  http://www.midnight-commander.org/ticket/2662
```
@tarruda Does libtermkey or whatever report mouse coordinates using the "new way" described at http://www.midnight-commander.org/ticket/2662 ? (It's a well-written ticket worth reading, could save us time)

Pyrohh replied:
As far as I'm aware, yes.

https://github.com/neovim/neovim/issues/1845#issuecomment-70988090
https://github.com/neovim/neovim/issues/1862#issuecomment-70988055

justinmk replied:
:+1: That answers it

justinmk commented on runtime/doc/various.txt:
```diff
@@ -277,6 +277,7 @@ g8			Print the hex values of the bytes used in the
 <			The screen is not redrawn then, thus you have to use
 			CTRL-L or ":redraw!" if the command did display
 			something.
+			Also see |:terminal|.
```
This makes more sense on line 236. I suggest this edit:

      :!{cmd}			Execute {cmd} with 'shell'. See also |:terminal|.

Pyrohh replied:
:+1:

justinmk commented on src/nvim/screen.c:
```diff
@@ -6450,7 +6450,6 @@ int screen_ins_lines (
   // - there is no valid screen
   // - the screen has to be redrawn completely
   // - the line count is less than one
```
this comment doesn't match the condition. And the entire comment is kind of redundant and/or outdated at this point.

Pyrohh replied:
No clue if it's outdated, but it does seem redundant. I removed it.

justinmk commented on src/nvim/option_defs.h:
```diff
@@ -567,21 +567,6 @@ EXTERN char_u   *p_titlestring; /* 'titlestring' */
 EXTERN long p_ttm;              /* 'ttimeoutlen' */
-EXTERN long p_ttyscroll;        /* 'ttyscroll' */
-#if defined(FEAT_MOUSE) && defined(UNIX)
-EXTERN char_u   *p_ttym;        /* 'ttymouse' */
```
@tarruda can you confirm that we're not going to use `'ttymouse'` and `'ttyscroll'` in the future?

Pyrohh replied:
@tarruda now that you're back, could you please answer this :-)

justinmk commented on src/nvim/option.c:
```diff
@@ -1581,13 +1581,12 @@ static vimoption_T
    {(char_u *)TRUE, (char_u *)0L} SCRIPTID_INIT},
+
+   // NO-OP
   {"ttymouse",    "ttym", P_STRING|P_NODEFAULT|P_NO_MKRC|P_VI_DEF,
```
we should probably remove it, else we will get more tickets asking why ttymouse works in Vim but not Neovim. Already fielding multiple such issues for `t_Co` and friends.

Pyrohh replied:
:+1: I hadn't considered that.

Commit:
Cleanup misc.

Regarding debugger.txt (which was Spotted by @Hettomei):
The third section was empty, and the second section is very outdated.
Nvim doesn't have things like Balloon Evalutation and Sun Visual
workshop integration, so just remove the section.

Regarding everything else:
- term.[ch] and term_defs.h don't exist anymore, so remove refs to them
- Add ttybuiltin to vim_diff.txt. It should have been done before, but
  vim_diff.txt didn't exist when ttybuiltin was removed (done in
  3baba1e7bc6698e6bc9f1d37fce88b30d6274bc9,)

Helped-by: Justin M. Keyes <justinkz@gmail.com>

Commit:
Cleanup 'weirdinvert' remnants

Removed in 3baba1e7bc6698e6bc9f1d37fce88b30d6274bc9, but the docs
weren't updated.

Commit:
Cleanup 'ttyscroll' remnants

Removed in 3baba1e7bc6698e6bc9f1d37fce88b30d6274bc9, but the docs
weren't updated.

Commit:
Remove `-T` command-line option

It didn't actually do anything after
3baba1e7bc6698e6bc9f1d37fce88b30d6274bc9

Helped-by: Justin M. Keyes <justinkz@gmail.com>

Commit:
Cleanup 'ttym[ouse]', FEAT_MOUSE, mouse_(dec|gpm|etc.)

Because of 3baba1e7bc6698e6bc9f1d37fce88b30d6274bc9, it's dead code.

Helped-by: Justin M. Keyes <justinkz@gmail.com>

Pyrohh commented:
This has been stagnating for a while because I haven't bothered dealing with the last commit (which was regarding 2-3 options). I've narrowed that down to just removing `-T` and marked as RFC.

Pyrohh commented:
Fixed the merge conflict and rebased. Could you take a look at this @glts?

glts commented:
I will after a night's rest :)

glts commented:
Just took a quick look. I can't really comment on this pull request as I have no idea about Neovim's tty stuff or mouse handling and I'm not in the loop about these 'terminal remnants', sorry.

Pyrohh commented:
> I can't really comment on this pull request as I have no idea about Neovim's tty stuff or mouse handling and I'm not in the loop about these 'terminal remnants', sorry.

It's fine, I appreciate you taking a look anyways.

@justinmk Sorry to keep doing this, but could you take a look? Not many people seem to be aware of the legacy terminal infrastructure.

Pyrohh commented:
Added one more small commit as discussed in https://github.com/neovim/neovim/pull/2609#discussion_r29975613.

Pyrohh commented:
Fixed the merge conflict and rebased.

Pyrohh commented:
I added more info to all the commit messages and repushed. Hopefully that makes reviewing this easier.

Pyrohh commented:
Rebasing changes to `runtime/syntax/vim.vim` is a nightmare due to the (very) long lines, so I split those changes out in to one commit.

@justinmk @fwalch could you take a look please?

justinmk commented:
Looking now

justinmk commented on runtime/doc/options.txt:
```diff
@@ -6771,80 +6771,16 @@ A jump table for the options with a short description can be found at |Q_op|.
-'ttymouse' 'ttym'	string	(default depends on 'term')
-			global
-			{only in Unix, doesn't work in the GUI; not
-			available when compiled without |+mouse|}
```
This ("only on unix, not in the GUI") is a good hint that it's not an option we want to keep. 

Platform-dependent features should be implemented as providers and/or passive events, possibly with lowest-common denominator support existing in the core (as is the case with TUI mouse handling).

Pyrohh replied:
:+1:

justinmk commented on runtime/doc/starting.txt:
```diff
@@ -772,7 +765,7 @@ To change it later, use a ":let" command like this: >
 greps in the help files) you might be able to use this: >
 
-	VIMRUNTIME=`vim -e -T dumb --cmd 'exe "set t_cm=\<C-M>"|echo $VIMRUNTIME|quit' | tr -d '\015' `
+	VIMRUNTIME=`vim -e --cmd 'exe "set t_cm=\<C-M>"|echo $VIMRUNTIME|quit' | tr -d '\015' `
```
This can be:

    nvim -e --cmd 'echo $VIMRUNTIME|quit' | tr -d '\015'

Pyrohh replied:
:+1: Although that didn't work for me because it was sending it to standard error. See the new commit.

justinmk commented:
LGTM, try squashing to 1 or 2 commits though.

Pyrohh commented:
Mkay, got it from 8 to 5 commits, with the only difference in the commit messages and that example you commented on.

I'll merge after the tests.
